libraries/helpers.rb

```diff
@@ -84,6 +91,70 @@ def osl_homeserver_db_defaults(pg_info)
         end
       end
 
+      # Generate the default configuration for matrix-appservice-irc
+      def osl_matrix_irc_defaults(custom_config)
+        default_config = {
+          'homeserver' => {
+            'url' => 'http://synapse:8008',
+            'domain' => new_resource.host_domain,
+          },
+          'ircService' => {
+            'passwordEncryptionKeyPath' => '/data/passkey.pem',
+            'mediaProxy' => {
+              'signingKeyPath' => '/data/signingkey.jwk',
+              'ttlSeconds' => 3600,
+              'bindPort' => 11111,
+              'publicUrl' => 'http://irc-media-proxy-not-implemented.localhost',
+            },
+            'permissions' => {
+              new_resource.host_domain => 'admin',
+            },
+          },
+          'database' => {
+            'engine' => 'postgres',
+            'connectionString' => "postgres://postgres:#{osl_matrix_genkey(new_resource.container_name)}@#{new_resource.container_name}-postgres:5432/postgres",
+          },
+        }.deep_merge(custom_config)
+
+        # Loop over all IRC servers, adding their defaults.
+        begin
+          custom_config['ircService']['servers'].each do |server, server_config|
+            default_config['ircService']['servers'][server] = osl_matrix_irc_server_defaults(server_config)
+          end
+        rescue
+          # No servers given
+        end
+
+        # Do a final merge
+        osl_yaml_dump(default_config)
+      end
+
+      # Generate the default configuration for every IRC server
+      def osl_matrix_irc_server_defaults(custom_config)
+        {
+          'name' => 'Untitled IRC Server',
+          'botConfig' => {
+            'enabled' => true,
+            'username' => 'oslmatrixbot',
+          },
+          'dynamicChannels' => {
+            'enabled' => true,
+            'published' => false,
+          },
+          'matrixClients' => {
+            'userTemplate' => '@as-irc_$NICK',
+          },
+          'ircClients' => {
+            'nickTemplate' => '$DISPLAY[OSL-Matrix]',
+            'kickOn' => {
+              'channelJoinFailure' => true,
+              'ircConnectionFailure' => true,
+              'userQuit' => true,
+            },
+          },
```

Comment:
Is this just a place holder or required? Can you please this a bit more?

Reply:
This is the bare minimum config needed for every IRC server we add. I've made this to act as the starting template to begin the config with. All attributes can, and will be, modified by whatever is passed into the `config` property.